crates/subspace-networking/src/create.rs

```diff
@@ -0,0 +1,187 @@
+use crate::behavior::Behaviour;
+use crate::kad::custom_record_store::CustomRecordStore;
+pub use crate::kad::custom_record_store::ValueGetter;
+use crate::node::Node;
+use crate::node_runner::NodeRunner;
+use crate::shared::Shared;
+use futures::channel::mpsc;
+use libp2p::dns::TokioDnsConfig;
+use libp2p::identify::{Identify, IdentifyConfig};
+use libp2p::kad::{Kademlia, KademliaBucketInserts, KademliaConfig, KademliaStoreInserts};
+use libp2p::noise::NoiseConfig;
+use libp2p::ping::Ping;
+use libp2p::swarm::SwarmBuilder;
+use libp2p::tcp::TokioTcpConfig;
+use libp2p::websocket::WsConfig;
+use libp2p::yamux::{WindowUpdateMode, YamuxConfig};
+use libp2p::{core, identity, noise, Multiaddr, PeerId, Transport, TransportError};
+use std::sync::Arc;
+use std::time::Duration;
+use std::{fmt, io};
+use thiserror::Error;
+
+const KADEMLIA_PROTOCOL: &[u8] = b"/subspace/kad";
+
+/// [`Node`] configuration.
+#[derive(Clone)]
+pub struct Config {
+    /// Identity keypair of a node used for authenticated connections.
+    pub keypair: identity::Keypair,
+    /// Nodes to connect to on creation.
+    pub bootstrap_nodes: Vec<(PeerId, Multiaddr)>,
+    /// List of [`Multiaddr`] on which to listen for incoming connections.
+    pub listen_on: Vec<Multiaddr>,
+    /// Adds a timeout to the setup and protocol upgrade process for all inbound and outbound
+    /// connections established through the transport.
+    pub timeout: Duration,
+    /// The configuration for the [`Kademlia`] behaviour.
+    pub kademlia_config: KademliaConfig,
+    /// Externally provided implementation of value getter for Kademlia DHT,
+    pub value_getter: ValueGetter,
+    /// Yamux multiplexing configuration.
+    pub yamux_config: YamuxConfig,
+    /// Should non-global addresses be added to the DHT?
+    pub allow_non_globals_in_dht: bool,
+    /// How frequently should random queries be done using Kademlia DHT to populate routing table.
+    pub initial_random_query_interval: Duration,
+}
+
+impl fmt::Debug for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("NetworkConfig").finish()
+    }
+}
+
+impl Config {
+    pub fn with_generated_keypair() -> Self {
```

Comment:
I might call it `with_random_keypair()`.

Reply:
I will think how API will evolve, in case we have more than one required field, this method will eventually go away.